pkg/osvscanner/vulnerability_result_internal_test.go

```diff
@@ -400,20 +413,113 @@ func Test_assembleResult(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name: "group vulnerabilities with license allowlist and license override",
+		args: args{
+			r:            &reporter.VoidReporter{},
+			packages:     packages,
+			vulnsResp:    vulnsResp,
+			licensesResp: makeLicensesResp(),
+			actions: ScannerActions{
+				ExperimentalScannerActions: ExperimentalScannerActions{
+					ShowAllPackages:       false,
+					ScanLicensesAllowlist: []string{"MIT", "0BSD"},
+				},
+				CallAnalysisStates: callAnalysisStates,
+			},
+			config: &config.ConfigManager{
+				OverrideConfig: &config.Config{
+					OverridePackageVersionLicenses: []config.OverridePackageVersionLicenseEntry{
+						{
+							Name:            "pkg-3",
+							Ecosystem:       "npm",
+							Major:           "1",
+							LicenseOverride: []string{"MIT"},
+						},
+					},
+				},
+			},
+		},
+		want: models.VulnerabilityResults{
+			ExperimentalAnalysisConfig: models.ExperimentalAnalysisConfig{
+				Licenses: models.ExperimentalLicenseConfig{
+					Allowlist: []models.License{models.License("MIT"), models.License("0BSD")},
+				},
+			},
+			Results: []models.PackageSource{
```

Comment:
For maintainability, can we just use the snapshot library here to store/match the results?

Reply:
should be done but check my code just in case I'm not understanding the snapshot library correctly